Turn function into arrow function

loucyx opened the pull request:
Using an arrow function instead of a function the source is smaller, and also the minified code:
```js
export default((a,b,c,d,e)=>{for(b=b.split?b.split("."):b,d=0;d<b.length;d++)a=a?a[b[d]]:e;return a===e?c:a});
```
I'll understand if there is a reason related to performance or similar to not use an arrow function, but I'm making a PR if you consider this to be a good change :smile:

Commit:
Turn function into arrow function

loucyx commented:
I also noticed that if we use `Array.prototype.some`, the loop runs until it finds a "falsy" value and then it stops, having less "looping", an even shorter source and minified code:
```js
export default (obj, path, def, undef) => {
	path = path.split ? path.split(".") : path;
	path.some(key => !(obj = obj ? obj[key] : undef));
	return obj === undef ? def : obj;
};
```
```js
export default((a,b,c,d)=>(b=b.split?b.split("."):b,b.some(b=>!(a=a?a[b]:d)),a===d?c:a));
```

Then again, maybe is a performance thing.

RReverser commented:
> Using an arrow function instead of a function the source is smaller, and also the minified code:

Hmm, did you use `npm run build`? It should also transpile code to ES5, while in your example it stays ES6 (hence the size difference).

loucyx commented:
@RReverser maybe making a PR to microbundle so the "module" build preserves the ES6 syntax instead of turning it into ES5? Also, even using microbundle, the other option I mentioned in https://github.com/developit/dlv/pull/32#issuecomment-495210223 has a smaller size:
```js
export default function(t,n,e,r){return(n=n.split?n.split("."):n).some(function(n){return!(t=t?t[n]:r)}),t===r?e:t}
```
And if you have and object like `{ foo: "bar" }` and you do `dlv(obj, 'foo.bar.foobar.0.whatever')` it will run twice instead of 5 times (thanks to the `some` method).

developit commented:
This library targets IE9+, so we can't rely on `.some()`. Modern output is [enabled by default](https://github.com/developit/microbundle/pull/413) in Microbundle 0.13 (in beta), which should automate the compression optimizations from this PR.

loucyx commented:
Sorry @developit, I never take IE into consideration. My bad! .... Closing then!
